app/src/main/java/com/example/leagueoflegends/presenter/MainActivityPresenter.kt

```diff
@@ -0,0 +1,25 @@
+package com.example.leagueoflegends.presenter
+
+import com.example.leagueoflegends.contract.BaseContract
+import com.example.leagueoflegends.misc.ApiService
+import com.example.leagueoflegends.misc.RetrofitProvider
+import com.example.leagueoflegends.model.Response
+import io.reactivex.Single
+
+class MainActivityPresenter: BaseContract.BasePresenter {
+    private var view: BaseContract.BaseView? = null
+    private val retrofit = RetrofitProvider.retrofit
+
+    override fun onAttach(view: BaseContract.BaseView) {
+        this.view = view
+    }
+
+    override fun onDetach() {
+        this.view = null
+    }
+
+    override fun getChampions(): Single<Response> {
+        return retrofit.create(ApiService::class.java).getChampions()
```

Comment:
creating a new ApiService every time is a lot of extra effort. I suggest putting this service into the DI too and let koin provide it. You don't need a new one each time anyway

Reply:
also, its not the responsibility of the presenter to create the connection to the API